guides/audio.md

```diff
@@ -0,0 +1,70 @@
+
+See the following example:
+
+```javascript
+a.setBins(5)
+osc(20,.1,2)
+	.saturate(()=>1-a.fft[4])
+	.rotate(()=>a.fft[0])
+	.kaleid()
+	.out()
+```
+
+See how if you make a deep "O" sound into the mic, the rotation will be strong and the saturation won't be affected as much. Also try to make a high "S" sound, you'll see the exact opposite.
+
+Note how we use brackets to call an element in an Array, and that we start counting from 0.
+
+#### a.bins & a.prevBins
+
+If you want the raw values from each bin without the mapping from 0 to 1, you can access them via  `a.bins`. You can also use `a.prevBins` to get the bins from the previous frame.
+
+### a.setSmooth()
+
+Sometimes the audio reactive elements react... too much. You can easily get into strobe territory if you are not careful. You can smooth out the interpretation of the sound using the `a.setSmooth()` function. A value of 0 will be no smoothing at all, a raw input, while a value of 1 will be so smooth nothing will happen at all. Try evaluating `a.setSmooth(.85)` above and see how different it looks.
+
+### a.setScale() & a.setCutoff()
+
+Each microphone, each sound input, etc, can be quite different in volume and dynamics. If you're on a noisy room, your visuals could react to the noise and that's quite annoying. Or if your mic is too low on volume, your visuals may barely react.
+If you ran `a.show()` and look at the graph, you may have noticed there are 2 horizontal lines going across the bins. The lowest one represents the cutoff (guitar players and other musicians out here might know this as a noise gate), this means that the value of the bin will be 0 unless the sound goes above that cutoff. The higher line is the scale, that's where the maximum value of 1 is (again, musicians may want to see this as a limiter with auto-gain). If a given bin goes above the scale value, its value won't go past 1.
+
+### a.vol
+
+`a.vol` will give you the overall volume of the audio input.
+
+### a.onBeat() & a.beat
+
+Hydra also has a simple beat detection algorithm. You can change this function to anything you like and it will be executed whenever Hydra detects a beat. The beat detection algorithm uses values from `a.vol` and compares them with a threshold set at `a.beat`.
+
+`a.beat` stores the configuration for the beat detection Hydra uses. It's an Object which most useful parameter is `threshold`. `a.beat.threshold` represents the volume to which `a.vol` will be compared to detect a beat. There's also `a.beat.decay` which sets the decay after a beat.
+
+### Reacting to music
+
+As we've seen, Hydra takes your microphone as an input, not your desktop audio. Those interested in using a music player or a DAW's output as an audio input will have to delve into virtual audio routing. Users with physical sound interfaces with multiple inputs and outputs might prefer physically routing an output to an input and set that input as the default microphone on Chrome.
```

Comment:
yes, i didn't take the time to add examples yet because they'd have to be links to the editor.
but i also wonder if there's a smart way of turning on the audio capture dynamically so that it only asks for microphone in this page.
i also experimented a bit trying to get docsify to load p5 only in the p5 guide

I'll keep experimenting tho !